### PR TITLE
Update module go.opentelemetry.io/contrib/config to v0.5.0

### PR DESCRIPTION
**Pull request by codeboten**
Replace https://github.com/open-telemetry/opentelemetry-collector/pull/9917 because of a merge conflict

### CODE REVIEW

File: config/configgrpc/configgrpc.go [hunk 1]
```diff
@@ -174,13 +174,13 @@ func (gcs *ClientConfig) isSchemeHTTPS() bool {
...
 // dial, use grpc.WithBlock() dial option.
-func (gcs *ClientConfig) ToClientConn(ctx context.Context, host component.Host, settings component.TelemetrySettings, extraOpts ...grpc.DialOption) (*grpc.ClientConn, error) {
+func (gcs *ClientConfig) ToClientConn(_ context.Context, host component.Host, settings component.TelemetrySettings, extraOpts ...grpc.DialOption) (*grpc.ClientConn, error) {
 	opts, err := gcs.toDialOptions(host, settings)
 	if err != nil {
 		return nil, err
 	}
 	opts = append(opts, extraOpts...)
-	return grpc.DialContext(ctx, gcs.sanitizedEndpoint(), opts...)
+	return grpc.NewClient(gcs.sanitizedEndpoint(), opts...)
```
Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:183]: Is this related to `go.opentelemetry.io/contrib/config`?

Comment (codeboten) [reply config/configgrpc/configgrpc.go:183]: Hmmm... in a previous version of this PR, an upgrade to a dep caused linting failures... hmmm... looks like in 1.63.2, the following deprecation notice was removed 🤔 

```
// Deprecated: use NewClient instead.  Will be supported throughout 1.x.
```

Comment (codeboten) [reply config/configgrpc/configgrpc.go:183]: See release notes, it was in .1: https://github.com/grpc/grpc-go/releases/tag/v1.63.1

Comment (codeboten) [reply config/configgrpc/configgrpc.go]: Split the change into a separate PR https://github.com/open-telemetry/opentelemetry-collector/pull/9965

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**